exporter/awsemfexporter/metric_translator.go

```diff
@@ -169,6 +205,25 @@ func (mt metricTranslator) translateOTelToGroupedMetric(rm pmetric.ResourceMetri
 	return nil
 }
 
+func fetchEntityFields(resourceAttributes pcommon.Map) (cloudwatchlogs.Entity, pcommon.Map) {
+	// the original resourceAttributes map is immutable, so we need to create a mutable copy
+	// to remove the entity fields from the attributes
+	mutableResourceAttributes := pcommon.NewMap()
+	resourceAttributes.CopyTo(mutableResourceAttributes)
+	serviceKeyAttr := map[string]*string{
```

Comment:
Name serviceKeyAttr is confusing. shouldn't this be keyAttributesMap?

Reply:
good point, I will rename that